backend/battling/views.py

```diff
@@ -68,8 +81,34 @@ def get_success_url(self):
 
 
 class DetailBattle(DetailView):
-    template_name = "battling/battle_details.html"
+    template_name = "battling/battle_detail.html"
     model = Battle
 
-    def get_battle(self):
-        return get_object_or_404(Battle, id=self.kwargs["pk"])
+    def get_object(self, queryset=None):
+        battle = get_object_or_404(Battle, id=self.kwargs["pk"])
+
+        return battle
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        battle = self.get_object()
+
+        creator = Team.objects.filter(battle=battle, trainer=battle.creator.id)
+        creator_pokemon = PokemonTeam.objects.filter(team=creator[0])
+
+        context["creator_team"] = [
+            creator_pokemon[0].pokemon,
+            creator_pokemon[1].pokemon,
+            creator_pokemon[2].pokemon,
+        ]
```

Comment:
I think this could be `context["creator_team"] = creator.pokemons.all()`. The following could be completely removed: `creator_pokemon = PokemonTeam.objects.filter(team=creator[0])`